src/main/java/org/simplify4u/plugins/ArtifactResolver.java

```diff
@@ -172,14 +172,24 @@ private Artifact resolveSignature(Artifact artifact, SignatureRequirement requir
     }
 
     private Artifact resolvePom(Artifact artifact) {
-        return resolve(repositorySystem.createProjectArtifact(
-                artifact.getGroupId(), artifact.getArtifactId(), artifact.getVersion()));
+        final Artifact pomArtifact = repositorySystem.createProjectArtifact(artifact.getGroupId(),
+                artifact.getArtifactId(), artifact.getVersion());
+        final ArtifactResolutionResult result = request(pomArtifact);
+        if (!result.isSuccess()) {
+            result.getExceptions().forEach(
+                    e -> log.debug("Failed to resolve pom " + pomArtifact.getId() + ": " + e.getMessage()));
+        }
+        return pomArtifact;
     }
 
-    private Artifact resolve(Artifact artifact) {
-        request(artifact);
-        // Evaluation of resolution results of all artifacts is done at a later stage,
-        // as resolution is performed in multiple stages.
+    private Artifact resolveArtifact(Artifact artifact) {
+        final ArtifactResolutionResult result = request(artifact);
+        if (!result.isSuccess()) {
+            result.getExceptions().forEach(e -> {
+                log.warn("Failed to resolve " + artifact.getId() + ": " + e.getMessage());
+                log.debug(e);
+            });
+        }
```

Comment:
code looks like will be  duplicated

Reply:
SonarCloud does not detect any duplication. I understand why you're saying that, but there's subtle differences. Personally, I wouldn't try to "deduplicate" this, as it will become hard to read with hardly any benefit.

The benefit will be a single line, as there are differences in treating dependency artifact and project artifacts as to avoid unnecessary flooding of uninteresting information. AFAICT it isn't worth it.

Feel free to propose something, though.

Reply:
@slawekjaranowski or would you want to change how we treat the _failure to resolve pom files_? If we treat those equal to dependency artifacts, then we can trivially reuse the same code. Right now I take the existing approach, which ignores poms that cannot be retrieved. I've been wondering whether or not that is the best choice of handling pom resolution errors.